lib/galaxy/webapps/galaxy/controllers/admin.py

```diff
@@ -455,13 +455,20 @@ class ToolVersionsColumn( grids.TextColumn ):
         def get_value( self, trans, grid, tool_version ):
             tool_ids_str = ''
             toolbox = trans.app.toolbox
-            for tool_id in tool_version.get_version_ids( trans.app ):
-                if toolbox.has_tool( tool_id, exact=True ):
-                    link = url_for( controller='tool_runner', tool_id=tool_id )
-                    link_str = '<a target="_blank" href="%s">' % link
-                    tool_ids_str += '<div class="count-box state-color-ok">%s%s</a></div><br/>' % ( link_str, tool_id )
-                else:
-                    tool_ids_str += '%s<br/>' % tool_id
+            tool = toolbox._tools_by_id.get(tool_version.tool_id)
+            if tool:
+                tool_id = tool.id
+                lineage = trans.app.toolbox._lineage_map.get(tool_id)
+                if lineage:
```

Comment:
If `lineage` is `None`, the method returns an empty string. I don't think that should ever happen, so it may be worth to remove this `if` and let an `Exception` be raised on the next line if something is really broken.